indexing-service/src/main/java/org/apache/druid/indexing/common/actions/TaskAction.java

```diff
@@ -38,9 +38,8 @@
     @JsonSubTypes.Type(name = "segmentTransactionalInsert", value = SegmentTransactionalInsertAction.class),
     @JsonSubTypes.Type(name = "segmentTransactionalAppend", value = SegmentTransactionalAppendAction.class),
     @JsonSubTypes.Type(name = "segmentTransactionalReplace", value = SegmentTransactionalReplaceAction.class),
-    // Type name doesn't correspond to the name of the class for backward compatibility.
+    @JsonSubTypes.Type(name = "segmentListById", value = RetrieveSegmentsByIdAction.class),
```

Comment:
I named it `segmentListById` to be consistent with `segmentListUsed` and `segmentListUnused`. If you feel strongly about this, I can include the rename in my follow up PR.

Reply:
The names are supposed to be consistent with the task action
The comment that was moved indicates that certain actions have a different key than the name because of backward compatibility.

Reply:
> The names are supposed to be consistent with the task action's.

Sure, this is preferable, but not a requirement.

In this case, it made more sense to me to adhere to the nomenclature that we are now going to support forever i.e. `segmentListXXX`. 

But I agree that it is better to stick to the convention used by all the other task actions rather than the 2 bad ones. Thanks for calling this out!